Tasks/JavaToolInstaller/FileExtractor/JavaFilesExtractor.ts

```diff
@@ -0,0 +1,145 @@
+        } else {
+            if (typeof this.xpSevenZipLocation == "undefined") {
+                this.xpSevenZipLocation = taskLib.which('7z', true);
+            }
+            return this.xpSevenZipLocation;
+        }
+    }
+
+    private async unzipExtract(file, destinationFolder) {
+        console.log(taskLib.loc('UnzipExtractFile', file));
+        if (process.platform == 'win32') {
+            // build the powershell command
+            let escapedFile = file.replace(/'/g, "''").replace(/"|\n|\r/g, ''); // double-up single quotes, remove double quotes and newlines
+            let escapedDest = destinationFolder.replace(/'/g, "''").replace(/"|\n|\r/g, '');
+            let command: string = `$ErrorActionPreference = 'Stop' ; try { Add-Type -AssemblyName System.IO.Compression.FileSystem } catch { } ; [System.IO.Compression.ZipFile]::ExtractToDirectory('${escapedFile}', '${escapedDest}')`;
+    
+            // change the console output code page to UTF-8.
+            let chcpPath = path.join(process.env.windir, "system32", "chcp.com");
+            await taskLib.exec(chcpPath, '65001');
+    
+            // run powershell
+            let powershell: tr.ToolRunner = taskLib.tool('powershell')
+                .line('-NoLogo -Sta -NoProfile -NonInteractive -ExecutionPolicy Unrestricted -Command')
+                .arg(command);
+            await powershell.exec();
+        }
+        else {
+            let unzip: tr.ToolRunner = taskLib.tool('unzip')
+                .arg(file);
+            await unzip.exec(<tr.IExecOptions>{ cwd: destinationFolder });
+        }
+    }
+
+    private async sevenZipExtract(file, destinationFolder) {
+        console.log(taskLib.loc('SevenZipExtractFile', file));
+        var sevenZip = taskLib.tool(this.getSevenZipLocation());
+        sevenZip.arg('x');
+        sevenZip.arg('-o' + destinationFolder);
+        sevenZip.arg(file);
+        var execResult = sevenZip.execSync();
+        if (execResult.code != taskLib.TaskResult.Succeeded) {
+            taskLib.debug('execResult: ' + JSON.stringify(execResult));
+        }
+    }
+
+    private async tarExtract(file, destinationFolder) {
+        console.log(taskLib.loc('TarExtractFile', file));
+        let tr: tr.ToolRunner = taskLib.tool('tar');
+        tr.arg(['xzC', destinationFolder, '-f', file]);
+        tr.exec();
+    }
+
+    private extractFiles(file: string, fileEnding: string) {
+        var stats = taskLib.stats(file);
+        if (!stats) {
+            throw new Error(taskLib.loc("ExtractNonExistFile", file));
+        } else if (stats.isDirectory()) {
+            throw new Error(taskLib.loc('ExtractDirFailed', file));
+        }
+
+        if (this.win) {
+            if (".tar" == fileEnding) { // a simple tar
+                    this.sevenZipExtract(file, this.destinationFolder);
+            } else if (".tar.gz" == fileEnding) { // a compressed tar, e.g. 'fullFilePath/test.tar.gz'
```

Comment:
you may also want to get in the habit of using single qotes instead of double quotes. It's just what most javascript devs use.